### PR TITLE
Implement QuickMenuStorageModel

### PR DESCRIPTION
**Pull request by jdragonbae**
This patch implements QuickMenuStorageModel. Furthermore, it will be
implemented to load and store data from persistent storage.

Refs: #958

### CODE REVIEW

File: src/org/triple/banana/quick_menu/QuickMenuStorageModelData.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package org.triple.banana.quick_menu;
+
+import androidx.annotation.IdRes;
+import androidx.annotation.StringRes;
+
+class QuickMenuStorageModelData extends QuickMenuViewModelData.ButtonInfo {
```
[the remark this answers is not in the document]
Comment (jdragonbae) [reply src/org/triple/banana/quick_menu/QuickMenuStorageModelData.java]: Thanks! Right, I agree with you. 
I'll update it.

File: src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java [hunk 2]
```diff
@@ -14,9 +14,11 @@
     public @NonNull QuickMenuController create(@NonNull Context context) {
         MockQuickMenuViewImpl mockView = new MockQuickMenuViewImpl(context);
         QuickMenuViewImpl view = new QuickMenuViewImpl(context);
+        QuickMenuStorageModel storageModel = new QuickMenuStorageModelImpl();
```
Comment (codeimpl) [inline src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java:17]: Do we need storageModel variable?

Comment (jdragonbae) [reply src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java:17]: Hmm.. I think so. Did you mean inject instance directly? 
like `new QuickMenuControllerImpl(..., new QuickMenuStorageModelImpl() ...);`
Am I right?

Comment (codeimpl) [reply src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java:17]: yes, I think we don't need the variable anymore.

Comment (romandev) [reply src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java:17]: @codeimpl, Why do you think of that? Do you think that it's better to reduce the stack usage here?
IMHO, in this case, using variable might be better for readability than reducing stack usage. WDYT?

Comment (codeimpl) [reply src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java:17]: I thought creating unused variables was not a good way, but in terms of readability, the current state seems to be better.

Comment (romandev) [reply src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java:17]: @codeimpl Sorry, I don't understand what you mean exactly. Do you mean the one-time used variable? (but you mentioned unused variable)

Comment (codeimpl) [reply src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java:17]: Oh sorry. I mean the one-time used variable right. : ) BTW, I agree with you.

**Review by romandev — COMMENTED** [review]
@jdragonbae Can you rebase this patch?
@codeimpl Is it okay to merge this as-is?

**Comment by codeimpl** [comment]
> @jdragonbae Can you rebase this patch?
> @codeimpl Is it okay to merge this as-is?

sure~

File: src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java (hunk 2 above)
Comment (jdragonbae) [reply src/org/triple/banana/quick_menu/QuickMenuFactoryImpl.java:17]: @romandev, @codeimpl Thanks for the review.

**Review by romandev — APPROVED** [review]
lgtm

**Pull request merged**